src/mono/mono/metadata/threads-types.h

```diff
@@ -580,4 +580,11 @@ mono_threads_summarize_execute (MonoContext *ctx, gchar **out, MonoStackHash *ha
 gboolean
 mono_threads_summarize_one (MonoThreadSummary *out, MonoContext *ctx);
 
+#if SIZEOF_VOID_P == 4
+/* Spin lock for unaligned InterlockedXXX 64 bit functions on 32bit platforms. */
+#define mono_interlocked_lock() mono_os_mutex_lock (&interlocked_mutex)
+#define mono_interlocked_unlock() mono_os_mutex_unlock (&interlocked_mutex)
+static mono_mutex_t interlocked_mutex;
+#endif
```

Comment:
A static var shouldn't be in a header file, every file including it will get a copy.

Reply:
Good catch! Made it to a global variable.